src/assets/TokenListController.ts

```diff
@@ -131,25 +126,29 @@ export class TokenListController extends BaseController<
     this.chainId = chainId;
     this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
     this.abortController = new AbortController();
-    onNetworkStateChange(async (networkState) => {
-      if (this.chainId !== networkState.provider.chainId) {
-        this.abortController.abort();
-        this.abortController = new AbortController();
-        this.chainId = networkState.provider.chainId;
-        if (this.state.preventPollingOnNetworkRestart) {
-          this.clearingTokenListData();
-        } else {
-          // Ensure tokenList is referencing data from correct network
-          this.update(() => {
-            return {
-              ...this.state,
-              tokenList: this.state.tokensChainsCache[this.chainId]?.data || {},
-            };
-          });
-          await this.restart();
+    this.messagingSystem.subscribe(
+      'NetworkController:providerChange',
```

Comment:
[Nit] What do you think about extracting this to a constant in NetworkController? This feels like the sort of knowledge that NetworkController should keep and expose to other places rather than copy around. Pro: more knowledge in one place. Con: Not sure if we've done that before for other controllers, so it would be a new pattern. Just a thought.

Reply:
yeah for sure! We export it already (just not directly) as a general pattern for V2 controllers from what I can tell ([see for example](https://github.com/MetaMask/controllers/pull/903/files#diff-62d026c0a52b005869c40865e71ae05c510b5f4fbd6d110ec75acefacd8dc727R88 ))

That said, I will leave it as is for this PR and make an issue to track making that change across all the controllers